samples/guestbook/gcp/provision_db/main.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The provision_db program connects to a Cloud SQL database and initializes it
+// with SQL from stdin. It's intended to be invoked from Terraform.
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+func main() {
+	if len(os.Args) != 1+5 {
+		fmt.Fprintf(os.Stderr, "usage: provision_db PROJECT SERVICE_ACCOUNT INSTANCE DATABASE ROOT_PASSWORD\n")
+		os.Exit(64)
+	}
+
+	log.SetPrefix("gcp/provision_db: ")
+
+	if err := provisionDb(os.Args[1], os.Args[2], os.Args[3], os.Args[4], os.Args[5]); err != nil {
+		log.Fatal(err)
+	}
+}
+
+type key struct {
+	PrivateKeyID string `json:"private_key_id"`
+}
+
+func provisionDb(projectID, serviceAccount, dbInstance, dbName, dbPassword string) error {
+	log.Printf("Downloading Docker images...")
+	mySQLImage := "mysql:5.6"
+	cloudSQLProxyImage := "gcr.io/cloudsql-docker/gce-proxy:1.11"
+	images := []string{mySQLImage, cloudSQLProxyImage}
+	for _, img := range images {
+		if _, err := run("docker", "pull", img); err != nil {
+			return err
+		}
+	}
+
+	log.Printf("Getting connection string from database metadata...")
+	gcp := &gcloud{projectID}
+	dbConnStr, err := run(gcp.cmd("sql", "instances", "describe", "--format", "value(connectionName)", dbInstance)...)
+	if err != nil {
+		return fmt.Errorf("getting connection string: %v", err)
+	}
+
+	// Create a temporary directory to hold the service account key.
+	// We resolve all symlinks to avoid Docker on Mac issues, see
+	// https://github.com/google/go-cloud/issues/110.
+	serviceAccountVoldir, err := ioutil.TempDir("", "guestbook-service-acct")
+	if err != nil {
+		return fmt.Errorf("creating temp dir to hold service account key: %v", err)
+	}
+	if err := os.Chdir(serviceAccountVoldir); err != nil {
+		return fmt.Errorf("changing to temp dir: %v", err)
+	}
+	defer os.RemoveAll(serviceAccountVoldir)
+	log.Printf("Created %v", serviceAccountVoldir)
+
+	// Furnish a new service account key.
+	if _, err := run(gcp.cmd("iam", "service-accounts", "keys", "create", "--iam-account="+serviceAccount, serviceAccountVoldir+"/key.json")...); err != nil {
+		return fmt.Errorf("creating new service account key: %v", err)
+	}
+	keyJSONb, err := ioutil.ReadFile(serviceAccountVoldir + "/key.json")
```

Comment:
Should be using `filepath.Join`. (Check elsewhere, too.)

Reply:
Done.